python_ta/checkers/invalid_name_checker.py

```diff
@@ -0,0 +1,476 @@
+    error_msgs = []
+
+    if not _is_in_snake_case(name):
+        error_msgs.append(
+            f'Argument name "{name}" is not in snake_case format. Argument names should be '
+            f"lowercase, with words separated by underscores. A single leading "
+            f"underscore can be used to indicate that the argument is not being used "
+            f"but is still needed somehow."
+        )
+
+    return error_msgs
+
+
+def _check_typevar_name(_node_type: str, name: str) -> List[str]:
+    """Returns a list of strings, each detailing how `name` violates Python naming conventions for
+    type variable names.
+
+    Returns an empty list if `name` is a valid type variable name."""
+    error_msgs = []
+
+    if not _is_in_pascal_case(name):
+        error_msgs.append(
+            f'Type variable name "{name}" is not in PascalCase format. Type variable '
+            f"names should have the first letter of each word capitalized with no separation "
+            f"between each word."
+        )
+
+    return error_msgs
+
+
+def _check_type_alias_name(_node_type: str, name: str) -> List[str]:
+    """Returns a list of strings, each detailing how `name` violates Python naming conventions for
+    type alias names.
+
+    Returns an empty list if `name` is a valid type alias name."""
+    error_msgs = []
+
+    if not _is_in_pascal_case(name):
+        error_msgs.append(
+            f'Type alias name "{name}" is not in PascalCase format. Type alias names should '
+            f"have the first letter of each word capitalized with no separation "
+            f"between each word."
+        )
+
+    return error_msgs
+
+
+# Map each variable name type to its corresponding check
+NAME_CHECK = {
+    "module": _check_module_name,
+    "constant": _check_const_name,
+    "class": _check_class_name,
+    "function": _check_function_and_variable_name,
+    "method": _check_method_and_attr_name,
+    "attribute": _check_method_and_attr_name,
+    "argument": _check_argument_name,
+    "variable": _check_function_and_variable_name,
+    "class attribute": _check_method_and_attr_name,
+    "class constant": _check_const_name,
+    "type variable": _check_typevar_name,
+    "type alias": _check_type_alias_name,
+}
+
+
+class InvalidNameChecker(BaseChecker):
+    """A checker class to report on names that don't conform to Python naming conventions.
+
+    For the Python naming conventions, see https://peps.python.org/pep-0008/#naming-conventions.
+    """
+
+    name = "naming_convention_violation"
+    # Template of displayed message will include the following:
+    #   1. Describe the variable name type (e.g. `module`, `const`, etc.) and the name itself.
+    #   2. Explanation of how the name violates the naming convention and suggest a
+    #   correction(s), if appropriate.
+    msgs = {
+        "C9103": (
+            'Found %s name "%s" in violation of Python naming conventions. %s',
```

Comment:
In terms of the overall message: I like what you've done with the explanations, they're more detailed. Because of this, I don't think the first boilerplate sentence ("Found %s name ...") is necessary, since each message can be read standalone and clearly conveys the problem. By deleting the first sentence you'll make each error message a bit shorter, and therefore clearer.

Also, to make the above messages a bit clearer, change "is not in" to "should be in". For example, "Class name "{name}" should be in PascalCase format..."